### DISCUSSION TITLE
Global property variable

### DISCUSSION THREAD

**Discussion by Lasthy** [godotengine/godot-proposals#5186] [Engine Core]
The idea would be to be able to create variables, like shader globals, that could be used across multiple nodes, and modified like, for example:

I could make a Node2D, place a bunch of children nodes, put a global property variable (let's call it global_angle) on the Node2D rotation_degrees, and to it's children, we could place on the rotation_degrees the value of -global_angle.

Then, when we rotate Node2D with the rotation_degrees, the children would rotate in the other way, and would keep it's relative angle to the player/dev.

I know this can be achieved by code, but it isn't able to preview on editor.

If someone knows if it is possible to do it already on version 3.x I would be grateful if you shared with me.

**Comment by Zireael07** [comment]
Export a variable on parent node (this lets you "preview in editor"), access it from children, done.

**Reply by Lasthy** [reply]
But then how would I "link" the variable on the child node? I want to make the child update with the parent instantly, on the editor, without having to code on _process, or executing a scene.

**Comment by azur-wolve** [comment]
When its about to have "global" values as you describe it, 2 options that I'm aware of:

- use an Autoload as global singleton, to store the values there
- store the values on the parent node

An additional option would be use custom resources, but its better to research before its caveats.

To execute code in-editor use **_tool scripts_**; put `tool` keyword on the top of the script.
(tool scripts have their own limitations, better research first)
